{% extends "v2/base.html" %}
{% load static %}
{% block background %}
<div class = bg-image
     style="
          background-image: url({% static 'img/IMG_1058.png' %});
          height: 100vh;background-repeat: no-repeat; background-size: cover; background-position: center center; opacity: 40%; background-color: black;
        ">
</div>
{% endblock %}
{% block navitems %}
                        <li class="nav-item active">
                            <strong><a class="nav-link" href="information">Information <span class="sr-only">(current)</span></a></strong>
                        </li>
            <li class="nav-item active">
                            <strong><a class="nav-link" href="#">Dashboard <span class="sr-only">(current)</span></a></strong>
                        </li>
{% endblock %}

{% block pagecontent %}
    <div class="mask rgba-gradient d-flex justify-content-center align-items-center">


     <div class="container rounded-4" data-wow-delay="0.3s">
       <div class="accordion accordion-flush" id="accordionFlushExample" style="background: rgba(255,255,255, .2) ">
  <div class="accordion-item" style="background: rgb(255,255,255, .2) ">
    <h2 class="accordion-header" id="flush-headingOne">
      <button
        class="accordion-button collapsed"
        type="button"
        data-mdb-toggle="collapse"
        data-mdb-target="#flush-collapseOne"
        aria-expanded="false"
        aria-controls="flush-collapseOne"
        style="background: rgb(255,255,255, .2) "
      >
          <h3>Transcriptomics</h3>
      </button>
    </h2>
    <div
      id="flush-collapseOne"
      class="accordion-collapse collapse"
      aria-labelledby="flush-headingOne"
      data-mdb-parent="#accordionFlushExample"
    >
      <div class="accordion-body">
          Transcriptome data shows which genes are activated, and to what extent these genes are active
          at the time of measurement (Lowe et al., 2017).This can be used to answer specific questions.
          For example, analyzing the transcriptomic data for multiple samples of interest, makes it possible
          to see if there are changes or differences in gene expression. When there are changes in an environment,
          genes can be up- or down regulated. If this is done in a controlled environment, the changes in gene expression
          can be related to the changes in the environment. In this way, the influence of substances on
          organisms in an environment can be investigated. Transcriptomics data can also be used to compare
          change or differences in different tissues or time. (Lowe et al., 2017).
          The most common way to extract transcriptomic data is by RNA-sequencing. In the last decade the count of
          publications that involve transcriptomic data acquired with RNA-sequencing (RNA-seq) is grown expansional
          (Lowe et al., 2017). The biggest contributor to this exponential growth of publication that use transcriptomic
          data acquired by RNA-seq is because RNA-seq has a dynamic range in order of magnitude 5 over its competitors.
          Also, the cost of sequencing has decreased significand over the last decades and this change made it more available
          (Pevsner, 2015). Almost all tools that are available, are designed to run form a command-line interface, either in a
          Unix environment or within the R/Bioconductor statical environment (Lowe et al., 2017).
          As a possible side effect, this may cause individuals who have no experience or training in
          working with a command-line interface see it as a deterrence to start using these techniques.
          Witch could have as a result that a lot of people do not us these techniques.  This is unfortunate
          mainly because the value of these tools are becoming increasingly apparent (Beghini et al., 2021).
          The purpose of the BakedinBiobakery is to create a workflow that makes it possible for users that do
          not have any experience with a command-line interface to do a full transcriptomics data analysis based
          on functional profiling in metagenomics in the form of a Web-user-Interface (WUI).

          <br><br>

          <b>References</b>
          <br>
          <i>Beghini, F., McIver, L. J., Blanco-MÃ­guez, A., Dubois, L., Asnicar, F., Maharjan, S., Mailyan, A., Manghi, P., Scholz, M., Thomas, A. M., Valles-Colomer, M., Weingart, G., Zhang, Y., Zolfo, M., Huttenhower, C., Franzosa, E. A., & Segata, N. (2021). Integrating taxonomic, functional, and strain-level profiling of diverse microbial communities with biobakery 3. ELife, 10. https://doi.org/10.7554/eLife.65088</i>
          <br>
          <i>Lowe, R., Shirley, N., Bleackley, M., Dolan, S., & Shafee, T. (2017). Transcriptomics technologies. PLoS Computational Biology, 13(5). https://doi.org/10.1371/journal.pcbi.1005457</i>

      </div>
    </div>
  </div>


           <div class="accordion-item" style="background: rgb(255,255,255, .2) ">
    <h2 class="accordion-header" id="flush-headingTwo">
      <button
        class="accordion-button collapsed"
        type="button"
        data-mdb-toggle="collapse"
        data-mdb-target="#flush-collapseTwo"
        aria-expanded="false"
        aria-controls="flush-collapseTwo"
        style="background: rgb(255,255,255, .2) "
      >
          <h3>Biobakery</h3>
      </button>
    </h2>
    <div
      id="flush-collapseTwo"
      class="accordion-collapse collapse"
      aria-labelledby="flush-headingTwo"
      data-mdb-parent="#accordionFlushExample"
    >
      <div class="accordion-body">
               The biobakery github= <a href="https://github.com/biobakery/">https://github.com/biobakery/</a>
          <br>
                The biobakery/HumaNn3.0 github= <a href="https://github.com/biobakery/humann"> https://github.com/biobakery/humann</a>

      </div>
    </div>
  </div>

               <div class="accordion-item" style="background: rgb(255,255,255, .2) ">
    <h2 class="accordion-header" id="flush-headingTree">
      <button
        class="accordion-button collapsed"
        type="button"
        data-mdb-toggle="collapse"
        data-mdb-target="#flush-collapseTree"
        aria-expanded="false"
        aria-controls="flush-collapseTree"
        style="background: rgb(255,255,255, .2) ">
          <h3>Usages</h3>
      </button>
    </h2>
    <div
      id="flush-collapseTree"
      class="accordion-collapse collapse"
      aria-labelledby="flush-headingTree"
      data-mdb-parent="#accordionFlushExample"
    >
      <div class="accordion-body">

<h5>Adding your initials to db</h5>

I understand that you are trilled to use this web-application, but first we need you to add yourself as a user.
          This can be done in on the page <code>addUsers</code>. If it runs on your localhost you can find it under:
<code>
http://127.0.0.1:8000/biobakery/addUser
</code>
          You can also just go to <code>http://127.0.0.1:8000/biobakery/Home</code> and click on <i>Your initials
are not there?</i> in the sign in screen.
<br><br>
Now that you have added yourself to the database the application can be used.
<br><br>
          <i><b>note:</b> it can take a while before the pages are refreshed and your name shows op in the sing in screen.</i>
<br><br>
<h5>File layout</h5>

The raw input data that is accepted by the BakedinBiobakery is a has a specific construction shown in figure1. A few rules must be complied. Otherwise, the workflow would not work correctly. The first layer of tis construction exist of a folder that has a name representing the analysis. The second layer consist of a group of folders that are named after the samples. In the most outer layer, is filled with fastq   files. The important thing to realize is that this application uses forward and backward reads to analyze the samples. This means that each sample folder has 2 inner fastq documents, both these files have a name that compliments the samples folder name, but they have both a unique extension: R1 and R2. Because these data files often contain a few gigabits of data, they are compressed. This makes them smaller and therefore easer to work with.
<br><br>
<img src="../../static/img/flowchart/file_layout.png" alt="figure1">

<br>Figure1:This figure shows a folder structure that is needed for letting the SnakePipeMultiHumaNn work.<br><br>
<h5>Starting process</h5>

After you have singed in, you will be redirected to the upload screen. On this page
you can choose witch parameters you want to use in your pipeline.
<br><br>
          <i><b>note:</b> if you decide to not use the --fastqc-end and --fastqc-start as
              kneaddata variables it will not be possible to evaluate the reads</i>
<br><br>
After you have decided witch parameter you want to use, the pipeline will start.
There are two options: 1) you wil be redirected to a page where the reads can be evaluated. or 2) The complete pipeline will run and the follow text wil be shown:
"<b>Great your file is sent to the system,
when the chosen tool is finished with all the hard work
          you wil get a message.</b>"
<br><br>
          <i><b>note:</b> To make fully use of the application all parameters are recommended (only <code>--verdose</code> is really optional)</i>

      </div>
    </div>
  </div>


       </div>
     </div>
    </div>
{% endblock %}

{% block script %}
<script type="text/javascript" src="{% static 'js/mdb.min.js' %}"></script><!-- Custom scripts -->
<script type="text/javascript"></script>
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js" integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>


{% endblock %}
